src/components/PureVueChart.vue

```diff
@@ -37,10 +37,12 @@
           >{{ bar.valueNotInMotion }}</text>
           <g v-if="showXAxis">
             <text
-              :x="bar.midPoint"
-              :y="`${innerChartHeight + 14}px`"
-              text-anchor="middle"
-            >{{ dataLabels[bar.index] }}</text>
+              :x="bar.midPoint"
+              :y="`${innerChartHeight + 14}px`"
+              text-anchor="middle"
+            >
+              <slot :name='dataLabels[bar.index]'>{{ dataLabels[bar.index] }}</slot>
```

Comment:
Why named slots for each index? imho you should have only one slot that you could use to customize label
```suggestion
              <slot name='label'>{{ dataLabels[bar.index] }}</slot>
```
And use in app like this:
```
<template v-slot:label>
  <tspan v-if="bar.index == 'N'">hello</tspan>
  <tspan v-if="bar.index == 'NW'">these</tspan>
....
  <tspan v-if="bar.index == 'NE'">&#128526;</tspan>
</template>
```

Reply:
I don't think that would work, because you won't be able to access 'bar'. This would only be able to change all the labels, The slots have to be named in order to change each label individually.

Reply:
You can use slot-scope to pass it